### PR TITLE
Audio: Clean up processing function and coefficients of dcblock on reset.

### PR DESCRIPTION
**Pull request by eddy1021**
This PR contributes to issues #4488

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@cujomalainey pls merge if good.

File: src/audio/dcblock/dcblock.c [hunk 1]
```diff
@@ -382,6 +382,10 @@ static int dcblock_reset(struct comp_dev *dev)
 	comp_info(dev, "dcblock_reset()");
 
 	dcblock_init_state(cd);
+
+	cd->dcblock_func = NULL;
+	memset(cd->R_coeffs, 0, sizeof(cd->R_coeffs));
```
Comment (cujomalainey) [inline src/audio/dcblock/dcblock.c:387]: R coeff is the config passed in and should not be touched

**Review by cujomalainey — CHANGES_REQUESTED** [review]

**Comment by lgirdwood** [comment]
@eddy1021 can you delete the merge commit from the PR. Thanks !

**Comment by eddy1021** [comment]
Oops.... I totally mess up the PR. sorry about that

**Comment by eddy1021** [comment]
> @eddy1021 can you delete the merge commit from the PR. Thanks !

Fixed the PR now. Sorry about adding many people to the reviewers

**Review by cujomalainey — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Review by RanderWang — APPROVED** [review]

**Comment by lgirdwood** [comment]
IPC4 build fixes merged yesterday, unrelated to this PR.

**Pull request merged**